doc/src/bond_cpp.md

```diff
@@ -185,6 +185,59 @@ particular protocol reader:
 
 See example: `examples/cpp/core/marshaling`.
 
+Type evolution
+==============
+
+Bond does not use explicit versioning to deal with changes to schemas over time
```

Comment:
"changed to a type's schema"? We're being inconsistent about using the word "type" and "schema".